axon-framework/axon-framework-commands/infrastructure.md

```diff
@@ -414,59 +376,242 @@ public DisruptorCommandBus commandBus(TransactionManager txManager, AxonConfigur
+ 2. The `MetaDataRoutingStrategy` - uses a property defined during creation of this strategy to fetch the routing key from the `CommandMessage` its `MetaData`.
+ 3. The `ERROR` `UnresolvedRoutingKeyPolicy` - the **default fallback** that will cause an exception to be thrown when the routing key cannot be resolved from the given `CommandMessage`.
+ 4. The `RANDOM_KEY` `UnresolvedRoutingKeyPolicy` - will return a random value when a routing key cannot be resolved from the `CommandMessage`. 
+    This effectively means that those commands will be routed to a random segment of the command bus.
+ 5. The `STATIC_KEY` `UnresolvedRoutingKeyPolicy` - will return a static key \(named "unresolved"\) for unresolved routing keys. 
+    This effectively means that all those commands will be routed to the same segment, as long as the configuration of segments does not change.
 
-* [SpringCloud](../../extensions/spring-cloud.md) or 
-* [JGroups](../../extensions/jgroups.md).
+The `AnnotationRoutingStrategy`  and `MetaDataRoutingStrategy` are considered the full implementations to configure.
+The `ERROR`, `RANDOM_KEY` and `STATIC_KEY` are _fallback routing strategies_ that should be configured on the annotation or meta-data implementations.
+To get a grasp how these are constructed, consider the following sample:
 
-Configuring a distributed command bus can \(mostly\) be done without any modifications in configuration files.
+{% tabs %}
+{% tab title="AnnotationRoutingStrategy" %}
+```java
+// A custom annotation can be used to drive the AnnotationRoutingStrategy
+@interface CustomRoutingAnnotation {
+}
+// Constructs a AnnotationRoutingStrategy with a different annotation and fallback: 
+public RoutingStrategy routingStrategy() {
+    return AnnotationRoutingStrategy.builder()
+                                    .annotationType(CustomRoutingAnnotation.class)
+                                    .fallbackRoutingStrategy(UnresolvedRoutingKeyPolicy.STATIC_KEY)
+                                    .build();
+}
+```
+{% endtab %}
+
+{% tab title="MetaDataRoutingStrategy" %}
+```java
+// Constructs a MetaDataRoutingStrategy with the defined meta data key and a different fallback:
+public RoutingStrategy routingStrategy() {
+    return MetaDataRoutingStrategy.builder()
+                                  .metaDataKey("my-routing-key")
+                                  .fallbackRoutingStrategy(UnresolvedRoutingKeyPolicy.RANDOM_KEY)
+                                  .build();
+}
+```
+{% endtab %}
+{% endtabs %}
+
+Of course a custom implementation of the `RoutingStrategy` can also be provided when necessary
+When we need to deviate from the default `AnnotationRoutingStrategy`, we should configure it like so: 
+
+{% tabs %}
+{% tab title="Axon Configuration API" %}
+```java
+public class AxonConfig {
+    // ...  
+    public void configureRoutingStrategy(Configurer configurer, RoutingStrategy routingStrategy) {
```

Comment:
```suggestion
    public void configureRoutingStrategy(Configurer configurer, YourRoutingStrategy yourRoutingStrategy) {
```

I think I would make it explicitly that this is a custom routing strategy impl

Reply:
Agreed!